File: Algorithms/p198_House_Robber/p198_House_Robber_1.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res1 = 0
        res2 = 0
        for v in nums:
            res1, res2 = res2, max(res1+v,res2)
        return res2

File: Algorithms/p198_House_Robber/p198_House_Robber_1.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int res1 = 0;
        int res2 = 0;
        for(int i=0;i<nums.size();++i){
            int tmp = res2;
            res2 = max(res1+nums[i],res2);
            res1 = tmp;
        }
        return res2;
    }
};
